src/states/node.rs

```diff
@@ -2458,3 +2476,336 @@ fn create_first_section_info(public_id: PublicId) -> Result<SectionInfo, Routing
         err
     })
 }
+
+#[cfg(all(test, feature = "mock_parsec"))]
+mod tests {
+    use super::*;
+    use crate::cache::NullCache;
+    use crate::mock_crust::crust::Config;
+    use crate::mock_crust::{self, Network};
+    use crate::outbox::{EventBox, EventBuf};
+    use crate::state_machine::{State, StateMachine};
+    use crate::xor_name::XOR_NAME_LEN;
+    use utils::LogIdent;
+
+    const ACCUMULATE_VOTE_COUNT: usize = 3;
+    const NOT_ACCUMULATE_ALONE_VOTE_COUNT: usize = 2;
+
+    struct NoteUnderTest {
+        pub machine: StateMachine,
+        pub full_id: FullId,
+        pub other_full_ids: Vec<FullId>,
+        pub other_parsec_map: Vec<ParsecMap>,
+        pub ev_buffer: EventBuf,
+        pub section_info: SectionInfo,
+    }
+
+    impl NoteUnderTest {
+        fn make_node() -> NoteUnderTest {
+            let full_ids = (0..4).map(|_| FullId::new()).collect_vec();
+            let mut ev_buffer = EventBuf::new();
+
+            let prefix = Prefix::<XorName>::default();
+            let section_info = unwrap!(SectionInfo::new(
+                full_ids.iter().map(|id| *id.public_id()).collect(),
+                prefix,
+                iter::empty()
+            ));
+
+            let gen_pfx_info = GenesisPfxInfo {
+                first_info: section_info.clone(),
+                latest_info: SectionInfo::default(),
+            };
+
+            let full_id = full_ids[0].clone();
+            let machine = make_state_machine(&full_id, &gen_pfx_info, &mut ev_buffer);
+
+            let other_full_ids = full_ids[1..4].iter().cloned().collect_vec();
+            let other_parsec_map = other_full_ids
+                .iter()
+                .map(|full_id| ParsecMap::new(full_id.clone(), &gen_pfx_info))
+                .collect_vec();
+
+            NoteUnderTest {
+                machine,
+                full_id,
+                other_full_ids,
+                other_parsec_map,
+                ev_buffer,
+                section_info,
+            }
+        }
+
+        fn node_state(&self) -> &Node {
+            match *self.machine.current() {
```

Comment:
You added `StateMachine::node_state`. Why not use it here too?

Reply:
I thought I did, but I must have misplaced it :-)